save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/config/KafkaConfiguration.kt

```diff
@@ -0,0 +1,164 @@
+import org.apache.kafka.clients.consumer.CooperativeStickyAssignor
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.clients.producer.internals.DefaultPartitioner
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.StringSerializer
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory
+import org.springframework.kafka.core.*
+import org.springframework.kafka.listener.DeadLetterPublishingRecoverer
+import org.springframework.kafka.listener.SeekToCurrentErrorHandler
+import org.springframework.kafka.support.serializer.ErrorHandlingDeserializer
+import org.springframework.kafka.support.serializer.JsonDeserializer
+import org.springframework.kafka.support.serializer.JsonSerializer
+import java.util.*
+
+/**
+ * Kafka producer and consumer configuration
+ *
+ * @property kafkaProperties
+ */
+@Configuration
+@Profile("dev & kafka")
+internal class KafkaConfiguration(
+    private val kafkaProperties: KafkaProperties) {
+    /**
+     * @property requestTopicName
+     */
+    @Value("\${kafka.test.execution.request.topic.name}")
+    lateinit var requestTopicName: String
+
+    /**
+     * @property requestTopicNameDlt
+     */
+    @Value("\${kafka.test.execution.request.topic.name}.DLT")
+    lateinit var requestTopicNameDlt: String
+
+    /**
+     * @property responseTopicName
+     */
+    @Value("\${kafka.test.execution.response.topic.name}")
+    lateinit var responseTopicName: String
+
+    /**
+     * @property consumerGroup
+     */
+    @Value("\${spring.kafka.consumer.group-id}")
+    lateinit var consumerGroup: String
+
+    /**
+     * @return kafka producer properties
+     */
+    @Bean
+    fun producerConfig(): Map<String, Any?> {
+        val producerProps = HashMap(kafkaProperties.buildProducerProperties())
+        producerProps[ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG] = StringSerializer::class.java as Object
+        producerProps[ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG] = JsonSerializer::class.java as Object
+        producerProps[ProducerConfig.PARTITIONER_CLASS_CONFIG] = DefaultPartitioner::class.java as Object
+        return producerProps
+    }
+
+    /**
+     * @return kafka producer factory
+     */
+    @Bean
+    fun producerFactory(): ProducerFactory<Any, Any> = DefaultKafkaProducerFactory(producerConfig())
+
+    /**
+     * @return kafka template
+     */
+    @Bean
+    fun kafkaTemplate(): KafkaTemplate<Any, Any> = KafkaTemplate(producerFactory())
+
+    /**
+     * @return kafka consumer properties
+     */
+    @Bean
+    fun consumerConfig(): Map<String, Any?> {
```

Comment:
maybe move this property to property file and use `@ConfigurationProperties(prefix = "kafka.consumer")`?

Reply:
in properties it is not convenient to set full class names, also not good to expose properties which is not supposed to change without changing business logic. also you already can set spring kafka properties in application-kafka.properties file which will be loaded in kafkaProperties